pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -416,6 +425,10 @@ public Optional<Schema<?>> getReaderSchema() {
     // rather than null.
     @Override
     public byte[] getSchemaVersion() {
+        if (msgMetadata.hasSchemaId()) {
+            byte[] schemaId = msgMetadata.getSchemaId();
+            return (schemaId.length == 0) ? null : schemaId;
+        }
```

Comment:
We should not return schema id from this method, because existing applications might already assume the schema version is a long byte array (`LongSchemaVersion`)

Reply:
If the schema ID exists, it indicates that users use the external schema.

Reply:
+1 add a private method